File: go/customer_create_request.go
// this file is @generated
package meteroid

type CustomerCreateRequest struct {
	Alias *string `json:"alias,omitempty"`

	BillingAddress *Address `json:"billing_address,omitempty"`

	BillingEmail *string `json:"billing_email,omitempty"`

	ConnectedAccountId *string `json:"connected_account_id,omitempty"`

	Currency Currency `json:"currency"`

	// User-defined custom property values, keyed by definition `key`. Validated against the
	// tenant's `CUSTOMER` property definitions. Omit to leave unset.
	CustomProperties map[string]any `json:"custom_properties,omitempty"`

	CustomTaxes RequiredSlice[CustomTaxRate] `json:"custom_taxes"`

	// Free-text legal exemption mention surfaced on exempt invoices.
	ExemptionReason *string `json:"exemption_reason,omitempty"`

	InvoicingEmails RequiredSlice[string] `json:"invoicing_emails"`

	InvoicingEntityId *InvoicingEntityId `json:"invoicing_entity_id,omitempty"`

	// Preferred document language (e.g. `en-US`, `fr-FR`); overrides the invoicing entity default.
	// Unsupported languages fall back to `en-US` when rendering.
	InvoicingLanguage *string `json:"invoicing_language,omitempty"`

	IsTaxExempt *bool `json:"is_tax_exempt,omitempty"`

	Name string `json:"name"`

	Phone *string `json:"phone,omitempty"`

	ShippingAddress *ShippingAddress `json:"shipping_address,omitempty"`

	VatNumber *string `json:"vat_number,omitempty"`
}

File: python/meteroid/models/customer_create_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .address import Address
from .currency import Currency
from .custom_tax_rate import CustomTaxRate
from .customer_type import CustomerType
from .invoicing_entity_id import InvoicingEntityId
from .shipping_address import ShippingAddress


@dataclasses.dataclass
class CustomerCreateRequest(BaseModel):
    currency: Currency

    custom_taxes: t.List[CustomTaxRate]

    invoicing_emails: t.List[str]

    alias: t.Optional[str] = None

    billing_address: t.Optional[Address] = None

    billing_email: t.Optional[str] = None

    connected_account_id: t.Optional[str] = None

    custom_properties: t.Optional[t.Any] = None
    """User-defined custom property values, keyed by definition `key`. Validated against the
    tenant's `CUSTOMER` property definitions. Omit to leave unset."""

    customer_type: t.Optional[CustomerType] = None
    """`INDIVIDUAL` requires `first_name`, `last_name`, and a billing-address country."""

    exemption_reason: t.Optional[str] = None
    """Free-text legal exemption mention surfaced on exempt invoices."""

    first_name: t.Optional[str] = None

    invoicing_entity_id: t.Optional[InvoicingEntityId] = None

    invoicing_language: t.Optional[str] = None
    """Deprecated: use `preferred_locales`. Applied only when `preferred_locales` is absent."""

    is_tax_exempt: t.Optional[bool] = None

    last_name: t.Optional[str] = None

    legal_number: t.Optional[str] = None
    """BT-47 — the buyer's national register identifier (SIREN/SIRET, HRB)."""

    name: t.Optional[str] = None
    """Required for `COMPANY`. Ignored for `INDIVIDUAL`: derived from `first_name` + `last_name`."""

    phone: t.Optional[str] = None

    preferred_locales: t.Optional[t.List[str]] = None
    """Preferred document languages, most-preferred first (BCP-47 tags, e.g.
    `["fr-FR", "en"]`); overrides the invoicing entity default. The first one the
    renderer has a template for wins, so an unsupported entry alongside a supported
    one just falls through; a list of only unsupported ones is rejected."""

    shipping_address: t.Optional[ShippingAddress] = None

    vat_number: t.Optional[str] = None
